src/Validator/WebhookValidator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace ZfrShopify\Validator;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Diactoros\PhpInputStream;
+use ZfrShopify\Exception;
+
+/**
+ * Validate an incoming webhook request coming from Shopify
+ *
+ * @author Michaël Gallego
+ */
+class WebhookValidator
+{
+    /**
+     * @link  https://docs.shopify.com/api/webhooks/using-webhooks#verify-webhook
+     * @param ServerRequestInterface $request
+     * @param string                 $sharedSecret Application shared secret to validate against
+     */
+    public function validate(ServerRequestInterface $request, string $sharedSecret)
+    {
+        $hmac = $request->getHeaderLine('X-Shopify-Hmac-Sha256');
+
+        if (empty($hmac)) {
+            throw new Exception\InvalidRequestException('Incoming Shopify webhook could not be validated');
+        }
+
+        $data           = new PhpInputStream();
```

Comment:
Does it do the same thing? I must have done this for a reason... but I don't remember :p